[cinatra, cmdline] Add new ports

EmployeeNo427 opened the pull request:
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [x] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html).
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [x] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

EmployeeNo427 committed:
Add new ports for cinatra and cmdline

EmployeeNo427 commented:
@microsoft-github-policy-service agree

LilyWangLL commented:
Please fix the CI pipeline error as below:
```
##[error]versions/c-/cinatra.json(2,): error : cinatra is missing a version database file at version/c-/cinatra.json
Run:
vcpkg x-add-version cinatra
to create the versions file.
##[error]versions/c-/cmdline.json(2,): error : cmdline is missing a version database file at version/c-/cmdline.json
Run:
vcpkg x-add-version cmdline
to create the versions file.
```

LilyWangLL commented on ports/cinatra/vcpkg.json:
```diff
@@ -0,0 +1,12 @@
+{
+    "name": "cinatra",
+    "version-string": "0.9.1",
```
```suggestion
    "version": "0.9.1",
```

LilyWangLL commented on ports/cinatra/portfile.cmake:
```diff
@@ -0,0 +1,24 @@
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO qicosmos/cinatra
+    REF 0.9.1
```
```suggestion
    REF ${VERSION}
```

LilyWangLL commented on ports/cinatra/portfile.cmake:
```diff
@@ -0,0 +1,24 @@
+)
+
+vcpkg_install_cmake()
+file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/cinatra RENAME copyright)
```
```suggestion
vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/LICENSE")
```

LilyWangLL commented on ports/cinatra/portfile.cmake:
```diff
@@ -0,0 +1,24 @@
+file(REMOVE_RECURSE ${SOURCE_PATH}/.git)
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include)
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug)
+
+
+
+
+
+
```
```suggestion
```

LilyWangLL commented on ports/cmdline/vcpkg.json:
```diff
@@ -0,0 +1,7 @@
+{
+  "name": "cmdline",
+  "version-string": "2014.02.04",
```
```suggestion
  "version-date": "2014-02-04",
```

LilyWangLL commented on ports/cmdline/vcpkg.json:
```diff
@@ -0,0 +1,7 @@
+  "name": "cmdline",
+  "version-string": "2014.02.04",
+  "description": "A simple, header-only command line parser for C++.",
+  "homepage": "https://github.com/tanakh/cmdline",
+  "dependencies": []
```
```suggestion
  "homepage": "https://github.com/tanakh/cmdline",
  "license": "BSD-3-Clause"
```

LilyWangLL commented on ports/cmdline/portfile.cmake:
```diff
@@ -0,0 +1,15 @@
+
+file(INSTALL ${SOURCE_PATH}/cmdline.h
+     DESTINATION ${CURRENT_PACKAGES_DIR}/include)
+
+     
+
```
```suggestion
```

LilyWangLL commented on ports/cmdline/portfile.cmake:
```diff
@@ -0,0 +1,15 @@
+    HEAD_REF master
+)
+
+file(COPY ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/cmdline)
```
```suggestion
vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/LICENSE")
```

LilyWangLL commented on ports/cmdline/portfile.cmake:
```diff
@@ -0,0 +1,15 @@
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO tanakh/cmdline
+    REF master
```
```suggestion
    REF e4cd007fb8f0314002d9a5b4d82939106e4144e4
```

LilyWangLL commented on ports/cinatra/vcpkg.json:
```diff
@@ -0,0 +1,12 @@
+    "name": "cinatra",
+    "version-string": "0.9.1",
+    "description": "Cinatra is a high performance HTTP framework with modern C++ features.",
+    "homepage": "https://github.com/qicosmos/cinatra",
```
```suggestion
    "homepage": "https://github.com/qicosmos/cinatra",
    "license": "MIT",
```

LilyWangLL commented:
There is a port which name is [cmdline](https://aur.archlinux.org/packages/cmdline), please add the branch name before the port name, like `tanakh-cmdline`.

LilyWangLL commented:
Note: I will be converting your PR to draft status. When you respond, please revert to "ready for review". That way, I can be aware that you've responded since you can't modify the tags.

LilyWangLL commented:
Can you explain what the following port cinatra does? Why after the installation is complete, this port only installs two tools and puts them under folder include?
![image](https://github.com/microsoft/vcpkg/assets/94091114/51508cc1-26bd-48d3-8060-07f2b84a6dbd)

EmployeeNo427 committed:
Update ports/cinatra/vcpkg.json

Co-authored-by: Lily Wang <94091114+LilyWangLL@users.noreply.github.com>

EmployeeNo427 committed:
Update ports/cinatra/portfile.cmake

Co-authored-by: Lily Wang <94091114+LilyWangLL@users.noreply.github.com>

EmployeeNo427 committed:
Update ports/cinatra/portfile.cmake

Co-authored-by: Lily Wang <94091114+LilyWangLL@users.noreply.github.com>

EmployeeNo427 committed:
Update ports/cinatra/portfile.cmake

Co-authored-by: Lily Wang <94091114+LilyWangLL@users.noreply.github.com>

EmployeeNo427 committed:
Update ports/cmdline/vcpkg.json

Co-authored-by: Lily Wang <94091114+LilyWangLL@users.noreply.github.com>

EmployeeNo427 committed:
Update ports/cmdline/vcpkg.json

Co-authored-by: Lily Wang <94091114+LilyWangLL@users.noreply.github.com>

EmployeeNo427 committed:
Update ports/cmdline/portfile.cmake

Co-authored-by: Lily Wang <94091114+LilyWangLL@users.noreply.github.com>

EmployeeNo427 committed:
Update ports/cmdline/portfile.cmake

Co-authored-by: Lily Wang <94091114+LilyWangLL@users.noreply.github.com>

EmployeeNo427 committed:
Update ports/cmdline/portfile.cmake

Co-authored-by: Lily Wang <94091114+LilyWangLL@users.noreply.github.com>

EmployeeNo427 committed:
Update ports/cinatra/vcpkg.json

Co-authored-by: Lily Wang <94091114+LilyWangLL@users.noreply.github.com>

EmployeeNo427 committed:
Rename cmdline to tanakh-cmdline and format manifests

EmployeeNo427 commented:
> Can you explain what the following port cinatra does? Why after the installation is complete, this port only installs two tools and puts them under folder include? ![image](https://private-user-images.githubusercontent.com/94091114/328118532-51508cc1-26bd-48d3-8060-07f2b84a6dbd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQ5OTkxMzksIm5iZiI6MTcxNDk5ODgzOSwicGF0aCI6Ii85NDA5MTExNC8zMjgxMTg1MzItNTE1MDhjYzEtMjZiZC00OGQzLTgwNjAtMDdmMmI4NGE2ZGJkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTA2VDEyMzM1OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJmY2VlZGY2YTNmYzgyZGQ2NTFkYWEzZWEwOTMyMGQzNjNkYmRmZWZjMDJjNWNhNGMwOWMwODE5OTJiOTcxMGQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.5_uSBttg94wunUg_jgoFn2FPy4jOw1Zdxelva7n3D_U)

Sorry, there are some mistakes on my end in writing the portfile, and this is not the expected behavior; the library has not been installed correctly.

LilyWangLL commented:
> Please fix the CI pipeline error as below:
> 
> ```
> ##[error]versions/c-/cinatra.json(2,): error : cinatra is missing a version database file at version/c-/cinatra.json
> Run:
> vcpkg x-add-version cinatra
> to create the versions file.
> ##[error]versions/c-/cmdline.json(2,): error : cmdline is missing a version database file at version/c-/cmdline.json
> Run:
> vcpkg x-add-version cmdline
> to create the versions file.
> ```

After you commit the new change, please fix CI errpr by the above solution.

EmployeeNo427 committed:
Fix incorrect handling of executable and header file installations in Cinatra portfile

- Resolved issues in the Cinatra portfile where only two executables were being installed from the example and press tool directories.

- Modified the portfile script to handle the inclusion of executable files, header files, while ensuring asio and async-simple headers are excluded to prevent conflicts.

EmployeeNo427 committed:
Merge branch 'microsoft:master' into add-cinatra-port

EmployeeNo427 committed:
Fix incorrect handling of executable and header file installations in Cinatra portfile. Add version info for tanakh-cmdline.

EmployeeNo427 commented:
Hi! Thank you very much for your help! I encountered some issues with my setup that required me to start from scratch. I have opened a new PR for the issue:
[cinatra, cmdline] Add new ports #38712
